salt/states/boto_asg.py

```diff
@@ -625,6 +637,18 @@ def present(
     return ret
 
 
+def _determine_termination_policies(termination_policies, termination_policies_from_pillar):
+    '''
+    helper method for present.  ensure that termination_policies are set
+    '''
+    pillar_termination_policies = copy.deepcopy(
+        __salt__['config.option'](termination_policies_from_pillar, [])
+    )
+    if not termination_policies and len(pillar_termination_policies) > 0:
+        termination_policies = pillar_termination_policies
```

Comment:
I'm hesitant to change this not because I disagree with you, but because "copy; check copied value" is a pattern used in pretty much all the helper functions in this file. That being said, since this is a list and the others are dictionaries this may not even need a deepcopy. If we can use a shallow copy instead that would provide better performance and retain the common pattern.